### PR TITLE
MooseDocs update to allow including multiple markdown sources

### LINKED ISSUE #6907: Add docs on Adaptive Mesh Refinement
### Description of the enhancement or error report

No docs on AMR can be found on the wiki through a search for "AMR" or "adaptive".

People struggle with this feature.

### LINKED ISSUE #7892: Justify Tables
A page containing multiple tables should be justified. Example:

http://mooseframework.org/docs/moose_docs/site/content/framework/systems/Kernels/AnisotropicDiffusion/

### LINKED ISSUE #8329: Documentation System Improvements
- [ ] Make the new site look correct on mobile device
- [ ] Make the search bar actually search (@dschwen I would appreciate help with this, I can't figure it out.)
- [x] Make the copy code button better
- [ ] Create tests for all markdown extensions and latex conversion items
- [ ] Create link checker (this will likely need to be done after a merge, because some links might not exist until after a merge)
- [x] Clean up code in NavigationNode and utilize the same code path for latex/pdf
- [x] Prevent MathJax from working in code areas
- [ ] Add documentation
- [ ] Auto link should look relative to the md file first, then search the database
- [ ] Need to handle pulling in documentation from submodules in an intelligent and automatic manner

### LINKED ISSUE #8792: Avoid double listing of Postprocessors under UserObjects
### Description of the enhancement or error report
Currently all Postprocessors have two .md files in the documentation system and are listed twice on the doc site, under Postprocessors and again under UserObjects. Please avoid that.

### Rationale for the enhancement or information for reproducing the error
Makes the docs more confusing, and introduces duplication (two .md doc files to be maintained)

### Identified impact
Improved documentation

### LINKED ISSUE #9044: Duplicate listing of input files that use the current object more than once
### Description of the enhancement or error report
The documentation system automatically links all input files (tests, examples, ..) that use a given object to its documentation page. If the object is used multiple times the inout file shows up multiple times in that list.

### Rationale for the enhancement or information for reproducing the error
Makes the lists longer than necessary and adds no value.

### Identified impact
Documentation system bugfix

### LINKED ISSUE #9077: Create custom livereload Watcher object to auto load new markdown
### Description of the enhancement or error report
Currently, the live view of the docs will not re-load when a new markdown file is created. This can be solve but it will require a custom livereload.Watcher object that if the number of markdown files change then rebuild the entire page and re-add all the pages to the individual page builder.

### Rationale for the enhancement or information for reproducing the error
Having this would make writing docs easier.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
None

### LINKED ISSUE #9140: Moosedocs.py create pdf not working
### Description of the enhancement or error report
I don't know if it is just me, but when I try to build a pdf using `moosedocs.py` I get an error that an extension (`materialize`) could not be loaded. 

### Rationale for the enhancement or information for reproducing the error

The full error message I get is
```
(docs) SHAMMY-CL:docs gre462$ ./moosedocs.py latex examples/report.md 
Traceback (most recent call last):
  File "./moosedocs.py", line 21, in <module>
    sys.exit(main.run())
  File "/Users/gre462/Documents/MOOSE/projects/moose/python/MooseDocs/main.py", line 192, in run
    retcode = commands.latex(**options) #pylint: disable=assignment-from-none
  File "/Users/gre462/Documents/MOOSE/projects/moose/python/MooseDocs/commands/latex.py", line 137, in latex
    parser = MooseDocs.MooseMarkdown(extensions=config.keys(), extension_configs=config)
  File "/Users/gre462/Documents/MOOSE/projects/moose/python/MooseDocs/MooseMarkdown.py", line 46, in __init__
    extension_configs=extension_configs)
  File "/Users/gre462/.conda/envs/docs/lib/python2.7/site-packages/markdown/__init__.py", line 159, in __init__
    configs=kwargs.get('extension_configs', {}))
  File "/Users/gre462/.conda/envs/docs/lib/python2.7/site-packages/markdown/__init__.py", line 185, in registerExtensions
    ext = self.build_extension(ext, configs.get(ext, {}))
  File "/Users/gre462/.conda/envs/docs/lib/python2.7/site-packages/markdown/__init__.py", line 264, in build_extension
    module = importlib.import_module(module_name_old_style)
  File "/Users/gre462/.conda/envs/docs/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: Failed loading extension 'MooseDocs.extensions.materialize' from 'MooseDocs.extensions.materialize', 'markdown.extensions.MooseDocs.extensions.materialize' or 'mdx_MooseDocs.extensions.materialize'
```

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

### PR DESCRIPTION
**Pull request by aeslaughter**
* Updates MooseDocs to use --json
* Allows documentation from Apps to be included
* Remove duplicate docs for PPs and UOs
* Updates app_syntax extension to use a single command (`!syntax`)

This doesn't address all the issues from #8329, but I think that this eliminates the major concerns and short comings found by users. Therefore I am closing the issue and a bunch of others that have been addressed already or by this PR.

(closes #8329)
(closes #9140)
(closes #9077)
(closes #9044)
(closes #8792)
(closes #7892)
(closes #6907)

### CODE REVIEW

**Comment by dschwen** [comment]
> Job Precheck:linux-gnu on 7621a80 wanted to post the following:
> Warning: This PR changes repo size by 53.47 MiB.

Love that check @brianmoose , is it a new thing?

**Comment by brianmoose** [comment]
@dschwen Yes, I put that in after the "BISON incident".

**Comment by gardnerru** [comment]
HAHA! @acasagran @zahom @bwspenc every procedure change is a scar.

**Comment by aeslaughter** [comment]
I have an update for Mastodon ready for these changes.

File: python/MooseDocs/common/moose_docs_file_tree.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    Inputs:
+        files[set]:
+    """
+    for item in os.listdir(directory):
+
+        # Complete path to the directory item (path or filename)
+        path = os.path.join(directory, item)
+
+        # Move along if path not in list of filesg
```
Comment (brianmoose) [inline python/MooseDocs/common/moose_docs_file_tree.py]: files

File: python/MooseDocs/common/nodes.py [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+        return self.name
+
+    def findall(self, name='', filter_=None):
+        """
+        Locate nodes based on a filter. By
+
+        Args:
+            name[str]: (optional) When the 'filter_' options is not supplied, this name is used
+                       in the default filter (default: '').
+            filter[function]: A filter function, if it returns true keep the node.  If not supplied
```
Comment (brianmoose) [inline python/MooseDocs/common/nodes.py]: `filter_`

File: python/MooseDocs/extensions/app_syntax.py [hunk 3]
```diff
@@ -147,57 +129,100 @@ class MooseSyntaxBase(MooseMarkdownCommon, Pattern):
       yaml[MooseYaml]: The MooseYaml object for the application.
       syntax[dict]: A dictionary of MooseApplicatinSyntax objects.
     """
+    @staticmethod
+    def defaultSettings():
+        """Default settings for MooseSyntaxBase."""
+        settings = MooseMarkdownCommon.defaultSettings()
+        settings['actions'] = (True, "Enable/disable action syntax lookup (this is used for "
+                                     "shared syntax such as BCs/Pressue).")
```
Comment (brianmoose) [inline python/MooseDocs/extensions/app_syntax.py]: Pressure

File: python/MooseDocs/extensions/app_syntax.py [hunk 4]
```diff
@@ -147,57 +129,100 @@ class MooseSyntaxBase(MooseMarkdownCommon, Pattern):
...
+        settings['objects'] = (True, "Enable/disable MooseObject syntax lookup (this is used for "
+                                     "shared syntax such as BCs/Pressue).")
```
Comment (brianmoose) [inline python/MooseDocs/extensions/app_syntax.py]: Pressure

File: python/mooseutils/mooseutils.py [hunk 5]
```diff
@@ -134,6 +133,9 @@ def check_configuration(packages):
         print "The following packages are missing but required:"
         for m in missing:
             print ' '*4, '-', m
+        print 'It may be possible to install them using "pip", but you likely need to ' \
+              'the MOOSE environment package on your system.\n'
+        print 'Using pip:\n    pip install package-name-here --user'
```
Comment (brianmoose) [inline python/mooseutils/mooseutils.py]: Probably want to use `print()` here and elsewhere.

File: python/MooseDocs/commands/build.py [hunk 6]
```diff
@@ -177,41 +145,24 @@ def build(config_file=None, site_dir=None, num_threads=None, no_livereload=False
...
-        """Builds complete documentation."""
-        builder = Builder(parser, site_dir)
-        builder.build(num_threads=num_threads)
-        return builder
-    builder = build_complete()
+    # Create the builder object and build the pages
+    builder = WebsiteBuilder(parser=parser, site_dir=site_dir, content=content)
+    builder.init()
+    if dump:
+        print builder
```
Comment (brianmoose) [inline python/MooseDocs/commands/build.py]: `print()`

File: python/MooseDocs/commands/check.py [hunk 7]
```diff
@@ -31,41 +39,40 @@ def check_options(parser):
...
-        yaml = mooseutils.MooseYaml(raw)
+    # Create the markdown parser and get the AppSyntaxExtension
+    config = MooseDocs.load_config(config_file, template=template, template_args=template_args)
+    parser = MooseMarkdown(config)
+    ext = parser.getExtension(AppSyntaxExtension)
+    syntax = ext.getMooseAppSyntax()
+
+    # Dump the complete syntax tree if desired
+    if dump:
+        print syntax
```
Comment (brianmoose) [inline python/MooseDocs/commands/check.py]: `print()`

**Review by brianmoose — CHANGES_REQUESTED** [review]
Some minor typos and print usage.
I didn't look through the second commit as it has 1600+ files and GitHub doesn't handle it well.
I hope you had a script to edit all those files....
I did go over the website and things looked fine.

**Review by brianmoose — APPROVED** [review]

**Pull request merged**